packages/blade/src/components/StepGroup/StepItem.tsx

```diff
@@ -0,0 +1,225 @@
+      ? props.theme.colors.interactive.background.primary.faded
+      : props.theme.colors.transparent,
+    borderRadius: props.theme.border.radius.medium,
+    width: '100%',
+    transition: `background-color ${props.theme.motion.duration.xquick} ${props.theme.motion.easing.standard.effective}`,
+    ':hover': {
+      backgroundColor: props.isSelected
+        ? props.theme.colors.interactive.background.primary.fadedHighlighted
+        : props.theme.colors.interactive.background.gray.fadedHighlighted,
+    },
+    ':focus-visible': {
+      ...getFocusRingStyles({ theme: props.theme }),
+    },
+  };
+});
+
+const getStepTypeFromIndex = ({
+  _index,
+  _nestingLevel,
+  itemsCount,
+}: GetStepTypeFromIndexProps): StepLineProps['stepType'] => {
+  if (_nestingLevel === 0) {
+    return 'default';
+  }
+
+  if (itemsCount === 1) {
+    return 'single-item';
+  }
+
+  if (_index === 0) {
+    return 'start';
+  }
+
+  if (_index === itemsCount - 1) {
+    return 'end';
+  }
+
+  return 'middle';
+};
+
+const _StepItem = ({
+  title,
+  timestamp,
+  description,
+  stepProgress = 'none',
+  marker,
+  trailing,
+  isSelected,
+  href,
+  target,
+  onClick,
+  children,
+  _index = 0,
+  _totalIndex = 0,
+  _nestingLevel = 0,
+}: StepItemProps): React.ReactElement => {
+  const {
+    itemsInGroupCount: itemsCount,
+    totalItemsInParentGroupCount,
+    orientation,
+    size,
+  } = useStepGroup();
+  const stepType = React.useMemo(
+    () => getStepTypeFromIndex({ _index, _nestingLevel, itemsCount }),
+    [_index, _nestingLevel, itemsCount],
+  );
+
+  const isFirstItem = _totalIndex === 0;
+  const isLastItem = _totalIndex === totalItemsInParentGroupCount - 1;
+  const isInteractive = Boolean(href) || Boolean(onClick);
+
+  if (trailing && orientation === 'horizontal') {
+    throwBladeError({
+      message: 'trailing is not allowed in horizontal StepGroup',
+      moduleName: 'StepItem',
+    });
+  }
+
+  const stepItemHeaderJSX = (
+    <Box display="flex" flexDirection="row" justifyContent="space-between" gap="spacing.4">
+      <Box>
+        <Text
+          size={stepItemHeaderTokens[size].title}
+          color="surface.text.gray.subtle"
+          weight="semibold"
+        >
+          {title}
+        </Text>
+        <Text
+          size={stepItemHeaderTokens[size].timestamp}
+          marginY="spacing.2"
+          color="surface.text.gray.muted"
+          variant="caption"
+        >
+          {timestamp}
+        </Text>
+        <Text size={stepItemHeaderTokens[size].description} color="surface.text.gray.muted">
+          {description}
+        </Text>
+      </Box>
+      <Box>{trailing}</Box>
+    </Box>
+  );
+
+  const stepItemHeaderPaddings: Omit<InteractiveItemHeaderProps, 'isSelected'> = {
+    paddingY: 'spacing.3',
+    paddingX: 'spacing.4',
+  } as const;
+
+  return (
+    <BaseBox
+      display="flex"
+      flexDirection={orientation === 'vertical' ? 'row' : 'column'}
```

Comment:
factor this out `const isVertical = orientation === 'vertical'` we are using it in many places